### ISSUE TITLE
L2 Network Created By Root user, Assigned To Specific Domain, Permissions not updated

### ISSUE THREAD

**Issue by midhunpjos** [apache/cloudstack#7569]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.17.2, 4.18
~~~

##### SUMMARY

Creating a L2 network by ROOT account, and assigning it to a specific Domain using Domain Selector does not create the network under the Selected Domain, Instead its created under ROOT domain only

##### STEPS TO REPRODUCE

~~~
1. Login As ROOT Admin user
2. Create L2 network
3. Select the Specific Domain and account on L2 network Creation
4. L2 network is creating under ROOT Domain Only, and the Specified Domain has no access to it
![image](https://github.com/apache/cloudstack/assets/82935957/8891daa4-ebab-40d9-afc9-72a4885dce46)
![image](https://github.com/apache/cloudstack/assets/82935957/7c6b312b-f0b9-4954-983f-1608ffae22a4)

~~~

##### EXPECTED RESULTS

~~~
L2 Network Need to be accessible by Domain Specified On Creation
~~~

##### ACTUAL RESULTS

~~~
Only ROOT Account  has access to the L2 Network


~~~

**Comment by harikrishna-patnala** [comment]
#7387 did not fix anything but made the tooltips more clear. I'll check the current issue here.

**Comment by DaanHoogland** [comment]
This seems to be a UI issue as the domain id and account name passed are not as the user specifies.
I will double check if this is an issue in the API but I don't think so @kiranchavala .

**Comment by DaanHoogland** [comment]
@midhunpjos , you can use the API as workaround. I am looking if I can fabrique a fix in the UI.

**Comment by DaanHoogland** [comment]
@harikrishna-patnala , you seem to have fixed this in #7393 , ready to ship in 4.18.1 

cc @midhunpjos , you can check that PR to verify.

**Comment by midhunpjos** [comment]
sure @DaanHoogland we can  close this issue
